client/lib/feed-stream-store/feed-stream.js

```diff
@@ -488,22 +500,27 @@ export default class FeedStream {
 
 		const postById = this.postById;
 		forEach( this.pendingPostKeys, function( postKey ) {
-			postById.add( postKey.postId );
+			postById.add( keyToString( postKey ) );
 		} );
 
 		const mostRecentPostDate = moment( this.postKeys[ 0 ][ this.dateProperty ] );
 
-		if ( this.pendingDateAfter > mostRecentPostDate ) {
+		// if the first element in the current postKeys isn't also in the pending set,
+		// we have a possible gap
+		const firstKey = this.postKeys[ 0 ];
+		if (
+			! some( this.pendingPostKeys, pendingKey => keysAreEqual( pendingKey, firstKey ) ) &&
+			this.pendingDateAfter > mostRecentPostDate
```

Comment:
i don't think i fully understand whats going on here.  so there are two conditions:
1. latest element of postKeys is not in the set just retrieved from the api
2. latest element of postKeys was posted before...... something. whats this one?